challenges/backend/src/app/services/CarOnSaleClient/classes/Calculator.ts

```diff
@@ -0,0 +1,37 @@
+import { injectable } from "inversify";
+import { AuctionItem } from "../dtos/auction/AuctionItem";
+
+@injectable()
+export class Calculator {
+    calculatePercentage(currentHighestBidValue: number, minimumRequiredAsk: number | null):number {
+        let auctionProgress = 0;
+        if (minimumRequiredAsk === null) {
+            return 0;
+        }
+
+        auctionProgress = currentHighestBidValue / minimumRequiredAsk! * 100;
+        return auctionProgress;
+    }
+    calculateAverageProgress(auctions: AuctionItem[]): number {
```

Comment:
The class method `calculateAverageProgress` must be marked either 'private', 'public', or 'protected'